### PR TITLE
Adding context and colors to CmdLineChecker

### PR DESCRIPTION
**Pull request by DKaman**
Hey Ryan!

I added some coloring functions and some other things to make the output of this script pop a bit :).

Most of the choices I made were purely aesthetic on my end so I'm up to change around some of that if it is deemed ugly lol. Let me know what you think!

### CODE REVIEW

**Comment by rfk** [comment]
Thanks @DKaman!  I probably wont get a chance to try this out for a day or two, but I look forward to taking it for a spin :-)

**Comment by DKaman** [comment]
No worries, lemme know what you think when you do :)

**Comment by rfk** [comment]
@DKaman I tried this out on the command-line, and I dig it!  Nice work.  I'm going to do a more detailed review of the code as well.

File: enchant/checker/CmdLineChecker.py [hunk 1]
```diff
@@ -55,95 +95,121 @@ class CmdLineChecker:
...
                 break
-        printf(["DONE"])
-    
+
+    def print_error(self):
+        """Prints the misspelled word along with 100 characters of
+        context on either side.  This number was arbitrarily chosen
+        and could be modified to be tunable or changed entirely.
+        It seems to be enough context to be helpful though
+        """
```
Comment (rfk) [inline enchant/checker/CmdLineChecker.py]: As purely a style nit, I usually try to format docstrings like this:

```
def whatever():
    """A short one-line summary of the thing.

    More text that describes the thing in detail
    and can go over several lines.
    """
```

File: enchant/checker/CmdLineChecker.py [hunk 2]
```diff
@@ -55,95 +95,121 @@ class CmdLineChecker:
...
-        
+
         if cmd == "i":
             return True
-        
+
         if cmd == "I":
             self.error.ignore_always()
             return True
-
```
Comment (rfk) [inline enchant/checker/CmdLineChecker.py]: Wow, I had a lot of indented blank lines in this file...

File: enchant/checker/CmdLineChecker.py [hunk 3]
```diff
@@ -42,9 +42,49 @@
...
+    """
+    Change text color for the Linux terminal.
+
+    Args:
+        string (str): String to colorify
+        color (str): Color to colorify the string in the following list:
+            black, red, green, yellow, blue, purple, cyan, gr[ae]y
+        prefix (str): Prefix to add to string (ex: Beginning of line graphics)
+    """
+    return colors[color] + prefix + string + colors['normal']
```
Comment (rfk) [inline enchant/checker/CmdLineChecker.py]: A nice addition to this function might to be check whether we're printing to a tty, and only add the colors if so.  Something like:

```
    if not sys.stdout.isatty():
        return prefix + string
    [... the bit that does the colouring ...]
```

File: enchant/checker/CmdLineChecker.py [hunk 4]
```diff
@@ -55,95 +95,121 @@ class CmdLineChecker:
...
+
     def run(self):
         """Run the spellchecking loop."""
         self._stop = False
         for err in self._checker:
             self.error = err
-            printf(["ERROR:", err.word.encode('utf8')])
-            printf(["HOW ABOUT:", err.suggest()])
+            self.print_error()
+            self.print_suggestions()
```
Comment (rfk) [inline enchant/checker/CmdLineChecker.py]: These helpers are much nicer than the previous stuff, thanks!

**Comment by DKaman** [comment]
@rfk thanks for the feedback! I'll get working on these things and push some changes soon. Will let you know when I do.

**Comment by DKaman** [comment]
Hey @rfk ! Sorry it's been so long, but I added the changes you've mentioned in the comments above. I am also currently working on a change that will help tune the context menu a bit more.

**Comment by rfk** [comment]
Thanks again @DKaman, I just re-discovered this PR in my inbox, and it looks great.  Merging.

**Pull request merged**